BTC Staking Form

desamtralized opened the pull request.

desamtralized committed:
port over staking component

desamtralized committed:
fix lint errors

desamtralized committed:
fix: more lint errors

desamtralized committed:
added main component holding wallet logic

desamtralized committed:
wallet provider

desamtralized committed:
staking component with contexts

desamtralized committed:
added finality providers

desamtralized committed:
showing stake preview

desamtralized committed:
fix lint error

desamtralized committed:
lint updates

desamtralized committed:
sending delegation tx

desamtralized committed:
linter fix

desamtralized committed:
wip: layout adjustments on staking form

sangeetparashar committed:
Complete delegation flow (#18)

* add changes to have walletContext used inside stake/btc and delegation information being passed

* lint fixes

* lint update

---------

Co-authored-by: Sangeet Parashar <sangeetparashar@gmail.com>

desamtralized committed:
new style for staking amount

desamtralized committed:
fix: remove unused values & import

desamtralized committed:
complete layout

desamtralized committed:
fix navigation

desamtralized committed:
redirecting to stake/btc after tx success

sangeetparashar committed:
Add current stake and staked btc if wallet is connected (#20)

* add current stake and staked btc

* lint fix

---------

Co-authored-by: Sangeet Parashar <sangeetparashar@gmail.com>

desamtralized committed:
remove unused components

sangeetparashar committed:
move getDelegations into its own hook and map pushTX for delegations (#21)

* move getDelegations into its own hook and map pushTX for delegations

* add ecc library and initialize before signing unbonding tx

---------

Co-authored-by: Sangeet Parashar <sangeetparashar@gmail.com>

MattNguyen commented on src/config/network.config.ts:
is there any benefit to calling this file `network.config.ts` vs `network.ts`

MattNguyen commented on src/utils/delegations/signStakingTx.ts:
```diff
@@ -1,16 +1,18 @@
 
 import { txFeeSafetyCheck } from "./fee";
-
+bitcoin.initEccLib(ecc);
```
might be useful to create a singleton instead of initing this everywhere it's used

desamtralized committed:
Rename network config and update Bitcoin library initialization

- Rename 'network.config.ts' to 'network.ts' and update imports across files
- Move Bitcoin library initialization to Wallet component useEffect
- Remove redundant Bitcoin library initialization from signUnbondingTx
- Update import statements in multiple files to reflect the new network config location

desamtralized committed:
connect wallet and staking form improvements

desamtralized committed:
refactor: improve staking form validation and UX

Implement debounced validation for staking amount and time inputs
Remove unnecessary wallet address dependency in Main component
Optimize error handling and state management in StakingAmount and StakingTime components
Replace lodash debounce with custom setTimeout implementation
Enhance form reset functionality and initial value handling
Improve accessibility by preventing default behavior on "Max" link click

desamtralized committed:
lint fix for pnpm-lock
